content/blog/2020/2020-4-015-covid-19-part-3.md

```diff
@@ -0,0 +1,112 @@
+{{% figure src="/blog_img/2020/2020-04-15-log-linechart.png"%}}
+We can now see first steps as well as current situation.
+
+### Python script
+We can see from these plots, that different countries came into contact with the virus at different times. It seems it would be
+much easier to compare the curves, if all of them started their climb at the same time. We can do that, with just a few lines of code. Example snippet here aligns the curves with the moment where the country recorded 100 cases. You can set n to any other (positive) number if you will though.
+
+```
+import numpy as np
+from copy import deepcopy
+
+n = 100
+out_data = deepcopy(in_data)
+out_data.X[out_data.X < n] = 0
+ar = np.argwhere(out_data.X)
+
+cols, shifts = np.unique(ar[:, 1], return_counts=True)
+out_data.X = np.array([np.roll(out_data.X[:, col], shift) for col, shift in zip(cols, shifts)]).T
+out_data.X[out_data.X==0]= np.nan`
+```
+Copy these lines into Pyscript widget, and plug in our initial snake of widgets like so:
+{{% figure src="/blog_img/2020/2020-04-15-snake-pyscript.png"%}}
+
+Look at the aligned curves now, they are much easier to compare.
+{{% figure src="/blog_img/2020/2020-04-15-script-log-lc.png"%}}
+
+
+### Absolute growth with derivative
+Let's take a look on how fast the confirmed cases are spreading. Connect As Timeseries with Difference widget, choose Differencing and select countries of interest. Differencing order of 1 means we'll be looking at derivative of first order, which just means daily change in our case. We'll leave shift as is, on 1.
+{{% figure src="/blog_img/2020/2020-04-15-diff-settings.png"%}}
+
+Again, we'll view the transformed curves in Line Chart. Notice how the difference transform (and every other) adds a new column. That means we can compare our curves in different states, so be sure to always check what is being shown. 
+For starters, try looking at for example China and it's transformed version. See how easy to spot daily spikes, that are perhaps out of the ordinary and need to be checked. The most prominent spike here is probably due to change in counting. 
+
+{{% figure src="/blog_img/2020/2020-04-15-china-dif.png"%}}
+
+If we again compare multiple countries, we notice the US's speedy climb and Iran seemingly being more successful in curbing the growth. 
+{{% figure src="/blog_img/2020/2020-04-15-diff-compare.png"%}}
+
+The difference could be only due to difference in population and what is a overwhelmingly large number of patients for one country might be almost business as usual for another one. 
+
+### Scale free growth with quotient
+Difference widget also has the option output coefficients. It will allow us to observe relative growth and compare countries directly.
```

Comment:
the option output? Probably *the output option*?

Reply:
Or better, *the option to output the coefficients*.

Reply:
Yeah, i was going for the latter :D